### PR TITLE
Add console command metadata

### PR DESCRIPTION
**Pull request by Cawllec**
Added ConsoleRequest class to ensure console command metadata is send to bugsnag: #211

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
None of this code is laravel specific actually, so could go into the php notifier?

File: src/Request/ConsoleRequest.php [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    protected $command;
+
+    /**
+     * Create a new console request instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        $this->command = $_SERVER['argv'];
```
Comment (GrahamCampbell) [inline src/Request/ConsoleRequest.php]: If we do want to go laravel specific, we should never access "super globals" in library files.

Comment (Cawllec) [inline src/Request/ConsoleRequest.php]: You're right, there's a more laravel way to do it with the Request object.  I'll check that the way to determine cli vs http in base PHP works across laravel and symfony and implement in the base notifier instead.  I'll close this for now.

Comment (GrahamCampbell) [inline src/Request/ConsoleRequest.php]: > You're right, there's a more laravel way to do it with the Request object.

You can actually use the symfony console application object. :)

Comment (GrahamCampbell) [inline src/Request/ConsoleRequest.php]: This is how Laravel handles commands internally.

Comment (Cawllec) [inline src/Request/ConsoleRequest.php]: In terms of generalising it as laravel (& symfony) uses it's own resolver object by default I think it'll be more appropriate to follow the existing pattern and make Laravel/Symfony/PHP specific console handlers with a look to generalising further down the line at some point

**Comment by Cawllec** [comment]
Re-opened as a laravel specific implementation of console metadata using Laravel objects.  Will look at generic and symfony specific console notifiers separately

**Comment by Cawllec** [comment]
Closing and refactoring to reflect [this change](https://github.com/bugsnag/bugsnag-php/pull/398)

**Comment by Cawllec** [comment]
Following the release of bugsnag-php v3.7.0 this request now uses the console request class from the main library to notify console issues.

**Comment by Cawllec** [comment]
I've added some safety when retrieving console args, ensuring the arguments are in a suitable format.

**Comment by GrahamCampbell** [comment]
Seems some extra commits got in here by mistake? Squash/rebase in order?

**Comment by Cawllec** [comment]
This should've gone out yesterday, I'll squash and merge this and release it today

**Comment by deminy** [comment]
Thanks for the updates; however, from what I have checked, the changes may not work, at least it doesn't work for our installation (Lumen 5.4 + [Dingo](https://github.com/dingo/api)).

Main difference between method **\Bugsnag\Request\BasicResolver::resolve()** and **\Bugsnag\BugsnagLaravel\Request\LaravelResolver::resolve()** is that:

* **BasicResolver::resolve()** uses **$_SERVER['argv']** to get command details.
* **LaravelResolver::resolve()** uses **\Illuminate\Http\Request->server('argv')** to get command details.

When running Artisan command, the **\Illuminate\Http\Request** request object exists in the application object but has no any request data populated in it, so **\Illuminate\Http\Request->$server**, **\Illuminate\Http\Request->$cookies**, etc are all empty.

The unit tests only check if return object is a **\Bugsnag\Request\ConsoleRequest** object or not; however, it doesn't tell if **$_SERVER['argv']** is passed in properly into that object or not.

I only checked/tested it under Lumen 5.4. I didn't check the changes with Laravel installation.

**Pull request merged**